### ISSUE TITLE
Look into supporting INLA

### ISSUE THREAD

**Issue by mjskay** [mjskay/tidybayes#223]
From a cursory glance at the R-INLA API the distribution functions for posterior marginals seem like one place to start: https://becarioprecario.bitbucket.io/inla-gitbook/ch-INLA.html#sec:marginals

Combining these with the new `stat_dist_...` family seems a logical step. Could also think about `point_interval`. Not sure what the equivalent of `fitted_draws()` and `predicted_draws()` would be in this case.

Would be happy if someone more familiar with INLA was interested in helping with this or had thoughts on what would be useful.

**Comment by mjskay** [comment]
Also look at inla.posterior.sample()

**Comment by tmeeha** [comment]
Hi Matthew. I use INLA a lot and would be happy to help with this. I am not a great package developer but can write R functions and know INLA posterior sampling well. Let me know if I can help.  Best, 
Tim

**Comment by mjskay** [comment]
I'd love that! I believe @dpsimpson had mentioned looking into it at some point but my guess is he had better things to do :)

There's a couple of natural directions to go I think...

1. Implementing `tidy_draws` for INLA so that spread_draws / gather_draws / etc work
2. Implementing `fitted_draws` and/or `predicted_draws`
3. Implementing something new for outputting analytical distributions. Something like a `tidy_dist`, `fitted_dist`, and/or `predicted_dist`, which would return distribution vectors that build off of the {distributional} package that could be visualized by the `stat_dist_` geoms (which I just recently moved to [ggdist](https://mjskay.github.io/ggdist/)). This is something I have had in the back of my mind for awhile but haven't built for any packages yet. If you're interested in that the relevant issue on ggdist to check out is here: https://github.com/mjskay/ggdist/issues/14

**Comment by dpsimpson** [comment]
Yes! I’ve been clobbered by other work. But I’m happy to chat with you if
you want a sounding board / rubber duck / or general INLA stuff

**Comment by tmeeha** [comment]
I have already written some clunky but functioning wrappers to inla.posterior.sample and can get the samples into an mcmc object. I am studying now how to work with mcmc objects in tidybayes right now. I have only used this with brm models in the past.

________________________________

**Comment by mjskay** [comment]
Cool! Thanks both. @tmeeha if you hit a snag figuring out stuff on the tidybayes side let me know

**Comment by becarioprecario** [comment]
Hi,

I use ggplot for all the figures in my INLA book but I believe that the code is very ad-hoc. I do not have any general functions but I will happy to help and test any code on the examples.

Best,

Virgilio
